src/main/java/ch/njol/skript/expressions/ExprPassengers.java

```diff
@@ -0,0 +1,197 @@
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Passenger")
+@Description({"The passenger of a vehicle, or the rider of a mob.",
+		"For 1.11.2 and above, it returns a list of passengers and you can use all changers in it.",
+		"See also: <a href='#ExprVehicle'>vehicle</a>"})
+@Examples({"#for 1.11 and lower",
+		"passenger of the minecart is a creeper or a cow",
+		"the saddled pig's passenger is a player",
+		"#for 1.11.2+",
+		"passengers of the minecart contains a creeper or a cow",
+		"the boat's passenger contains a pig",
+		"add a cow and a zombie to passengers of last spawned boat",
+		"set passengers of player's vehicle to a pig and a horse",
+		"remove all pigs from player's vehicle",
+		"clear passengers of boat"})
+@Since("2.0, 2.2-dev26 (Multiple passengers for 1.11.2+)")
+public class ExprPassengers extends SimpleExpression<Entity> { // SimpleExpression due to isSingle
+
+	static {
+		Skript.registerExpression(ExprPassengers.class, Entity.class, ExpressionType.PROPERTY,
+				"[the] passenger[:s] [of %entities%]", // Passenger can be non plural due to event default expression 'passenger'
+				"%entities%'[s] passenger[s]"
+		);
+	}
+
+	@Nullable
+	private Expression<Entity> vehicles;
+	private boolean plural;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		vehicles = (Expression<Entity>) exprs[0];
+		plural = parseResult.hasTag("s");
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected Entity[] get(Event event) {
+		Entity[] source = vehicles.getAll(event);
+		Converter<Entity, Entity[]> converter = new Converter<Entity, Entity[]>() {
+			@Override
+			@Nullable
+			public Entity[] convert(Entity entity) {
+				if (getTime() != EventValues.TIME_PAST && event instanceof VehicleEnterEvent && entity.equals(((VehicleEnterEvent) event).getVehicle()))
+					return new Entity[] {((VehicleEnterEvent) event).getEntered()};
+				if (getTime() != EventValues.TIME_PAST && event instanceof VehicleExitEvent && entity.equals(((VehicleExitEvent) event).getVehicle()))
+					return new Entity[] {((VehicleExitEvent) event).getExited()};
+				if (getTime() != EventValues.TIME_PAST && event instanceof EntityMountEvent && entity.equals(((EntityMountEvent) event).getEntity()))
+					return new Entity[] {((EntityMountEvent) event).getEntity()};
+				if (getTime() != EventValues.TIME_PAST && event instanceof EntityDismountEvent && entity.equals(((EntityDismountEvent) event).getEntity()))
+					return new Entity[] {((EntityDismountEvent) event).getEntity()};
+				return entity.getPassengers().toArray(new Entity[0]);
+			}};
+		List<Entity> entities = new ArrayList<>();
+		for (Entity entity : source) {
+			if (entity == null)
+				continue;
```

Comment:
`entity` cannot be null

Reply:
It can be null here, but that's only because the existing implementation uses getAll. I'll remove this and change getAll to getArray